Chargebee/Classes/Restore/RestoreError.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  RestoreError.swift
+//  Chargebee
+//
+//  Created by ramesh_g on 09/03/23.
+//
+
+
+import StoreKit
+
+
+public enum RestoreError: Error {
+    case noReceipt
+    case refreshReceiptFailed
+    case restoreFailed
+    case invalidReceiptURL
+    case invalidReceiptData
+    case noProductsToRestore
+}
```

Comment:
How does it work along side the current errors?

Reply:
I have added one more case to RestoreError to handle the network issue  
 case serviceError(error: String)
 completion?(.failure(.serviceError(error: error.localizedDescription)))


Only one API we have and it will  give back the exact error thrown by the server 
And rest all the cases are checking the local validations and placed in according to the error  
RestoreError  is similar to the existing  CBPurchaseError for handling nonserver errors (Local device based validations)

shown in the example project pushed changes

Reply:
for Storekit  Receipt Refresh failure case i updated RestoreError one enum case 
` case .refreshReceiptFailed(error: let errorMessage):`